README.md

```diff
@@ -86,53 +110,710 @@ ubuntu@infra1:~$ maas-anvil cluster list
 user: $user
 ```
 
-And `juju login` as usual.
+And use `juju login` as usual.
+
+## Configure your MAAS Anvil deployment
+
+When deploying MAAS in high availability, you may need to configure the maximum connection to the database, the virtual IP, TLS, the charms versions used or even the way a component is deployed. MAAS Anvil allows you to configure all of these things, and this section explains how to do it.
+
+If you want to know exactly what configuration options are available and what effects they have, please read the section on [Configuration options](#configuration-options).
+
+The configuration options of MAAS Anvil are generally divided into two categories:
+
+-   Deployment
+-   Software
+
+In the deployment category you can configure general options for deployment, in the software category you can select versions and configuration of all charms used in MAAS Anvil and define how these charms are deployed.
+
+### `--accept-defaults` flag
+
+If you set the `--accept-defaults` flag on the bootstrapped and joining nodes you will accept the default configuration that comes with MAAS Anvil. For all available configuration options you can see the default option in the [Configuration options](#configuration-options) section.
+
+### Configuration prompting
+
+If you omit the `--accept-defaults` flag, you will be prompted to enter the deployment configuration during deployment. However, you will not be prompted for the software configuration. The software default settings will still be selected.
+
+### Manifest file
+
+If you want to define the entire configuration of the MAAS Anvil deployment in advance, both deployment and software, you can do so with a manifest file. A manifest file is a yaml file that can be used to specify all configurations for a MAAS Anvil cluster deployment.
+
+Run the following command to generate a manifest file for MAAS Anvil:
+
+```bash
+maas-anvil manifest generate
+```
+
+A manifest file will be created in the default location `$HOME/.config/anvil/manifest.yaml`. If you have a running MAAS Anvil installation, the manifest file will be based on the configurations of your running MAAS Anvil cluster. If no bootstrap has been performed yet, you will receive a default configuration file that looks something like this:
+
+```yaml
+deployment:
+  bootstrap:
+    # Management networks shared by hosts (CIDRs, separated by comma)
+    management_cidr: ""
+  postgres:
+    # Maximum number of concurrent connections to allow to the database server
+    max_connections: "default"
+  haproxy:
+    # Virtual IP to use for the Cluster in HA
+    virtual_ip: ""
+    # The TLS mode
+    tls_mode: "disabled"
+    # Path to SSL Certificate for HAProxy (enter nothing to skip TLS)
+    ssl_cert: ""
+    # Path to private key for the SSL certificate (enter nothing to skip TLS)
+    ssl_key: ""
+    # Path to CA certificate, if you want to use a self-signed certificate when
+    # in passthrough mode
+    ssl_cacert: ""
```

Comment:
The manifest actually looks like this now:

```
  haproxy:
    # Virtual IP to use for the Cluster in HA
    virtual_ip: ""
    # Path to SSL Certificate for HAProxy
    ssl_cert: ""
    # Path to private key for the SSL certificate
    ssl_key: ""
    # Path to CA cert chain, for use with self-signed SSL certificates (enter nothing to skip)
    ssl_cacert: ""
    # TLS mode: ['termination', 'passthrough', 'disabled']?
    tls_mode: "disabled"
  maas-region:
    # Path to SSL Certificate for HAProxy
    ssl_cert: ""
    # Path to private key for the SSL certificate
    ssl_key: ""
    # Path to CA cert chain, for use with self-signed SSL certificates (enter nothing to skip)
    ssl_cacert: ""
    # TLS mode: ['passthrough', 'disabled']?
    tls_mode: "disabled"
```

I know the duplicate questions are confusing, but it's a limitation we have since we don't know what charms will be deployed when we generate a manifest. 

We should also add a note somewhere that if both haproxy and maas-region are to be installed, the answers will be taken from the haproxy section. If only maas-region is to be installed, the answers are taken from the maas-region section